server/router_utils.go

```diff
@@ -215,6 +215,47 @@ func readOrganizationIDs(writer http.ResponseWriter, request *http.Request) ([]t
 	return organizationsConverted, nil
 }
 
+func readRuleIDWithErrorKey(writer http.ResponseWriter, request *http.Request) (types.RuleID, types.ErrorKey, error) {
+	ruleIDWithErrorKey, err := getRouterParam(request, "rule_id")
+	if err != nil {
+		const message = "unable to get rule id"
+		log.Error().Err(err).Msg(message)
+		handleServerError(writer, err)
+		return types.RuleID(0), types.ErrorKey(0), err
+	}
+
+	splitedRuleID := strings.Split(string(ruleIDWithErrorKey), "|")
+
+	if len(splitedRuleID) != 2 {
+		err = fmt.Errorf("invalid rule ID, it must contain only rule ID and error key separated by |")
+		log.Error().Err(err)
+		handleServerError(writer, &RouterParsingError{
+			paramName:  "rule_id",
+			paramValue: ruleIDWithErrorKey,
+			errString:  err.Error(),
+		})
+		return types.RuleID(0), types.ErrorKey(0), err
+	}
+
+	IDValidator := regexp.MustCompile(`^[a-zA-Z_0-9.]+$`)
```

Comment:
is `0`, `.`, `-` accepted ID or error key?

Reply:
Yes, for ID mostly used `a-z` and `.`, in error key are mostly used `A-Z` and `_`, but it could also be `a-z` and etc.

Reply:
I mean, with the proposed regexp, even single `.` will be accepted. Or `..` etc.

Reply:
yes, and we use this pattern not only here, but in other places as well, to check ID. So I think it should be another refactor task